src/main/java/net/hypixel/nerdbot/util/gson/HypixelPlayerResponse.java

```diff
@@ -0,0 +1,151 @@
+package net.hypixel.nerdbot.util.gson;
+
+import com.google.gson.annotations.SerializedName;
+import lombok.Getter;
+import net.hypixel.nerdbot.util.Util;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class HypixelPlayerResponse {
+
+    private boolean success;
+    private Player player;
+
+    public boolean isSuccess() {
+        return success;
+    }
+
+    public Player getPlayer() {
+        return player;
+    }
+
+    public static class Player {
+
+        @SerializedName("_id")
+        @Getter private String hypixelId;
+        private String uuid;
+        @SerializedName("displayname")
+        @Getter private String displayName;
+        @SerializedName("channel")
+        @Getter private String chatChannel;
+        @Getter private Instant firstLogin;
+        @Getter private Instant lastLogin;
+        @Getter private Instant lastLogout;
+        @Getter private long networkExp;
+        @Getter private long karma;
+        @Getter private int achievementPoints;
+        @Getter private long totalDailyRewards;
+        @Getter private long totalRewards;
+        @Getter private String mcVersionRp;
+        @Getter private String mostRecentGameType;
+        @SerializedName("playername")
+        private String playerName;
+        @Getter private List<String> knownAliases;
+        @Getter private SocialMedia socialMedia;
+        private List<Object> achievementsOneTime;
+        private transient List<String> achievementsOneTimeFixed;
+        @Getter private String currentClickEffect;
+        @Getter private String currentGadget;
+        @SerializedName("claimed_potato_talisman")
+        @Getter private Instant claimedPotatoTalisman;
+        @SerializedName("skyblock_free_cookie")
+        @Getter private Instant skyblockFreeCookie;
+        @SerializedName("claimed_century_cake")
+        @Getter private Instant claimedCenturyCake;
+        @SerializedName("scorpius_bribe_120")
+        @Getter private Instant scorpiusBribe120;
+        @Getter private Map<String, Long> voting;
+        @Getter private Map<String, Integer> petConsumables;
+        @Getter private Map<String, Integer> achievements;
+        @Getter private Map<String, Instant> achievementRewardsNew;
+
+        // Rank
+        private String packageRank;
+        private String newPackageRank;
+        private String monthlyPackageRank;
+        private String rank;
+        private String prefix;
+        private String monthlyRankColor;
+        private String rankPlusColor;
+        private String mostRecentMonthlyPackageRank;
```

Comment:
Can we not simplify this into one field for rank and plus colour?

Reply:
All those fields exist, and it requires a complex method to determine someone's current rank, that method was removed and this data may as well exist in the converted class even if nothing is done with it